[GTK] Use the new GTK API to create a GdkTexture from a DMA-BUF buffer

carlosgcampos opened the pull request:
#### a07d3e4e2067578a818215d89f5742b5e9707389
<pre>
[GTK] Use the new GTK API to create a GdkTexture from a DMA-BUF buffer
<a href="https://bugs.webkit.org/show_bug.cgi?id=268445">https://bugs.webkit.org/show_bug.cgi?id=268445</a>

Reviewed by Alejandro G. Castro.

Using GdkDmabufTextureBuilder we can create a GdkTexture to pass to GTK
without having to use EGL.

* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::create):
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::BufferDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::didUpdateContents):
(WebKit::AcceleratedBackingStoreDMABuf::Renderer::snapshot const):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBuffer):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.h:

Canonical link: <a href="https://commits.webkit.org/273890@main">https://commits.webkit.org/273890@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7df17da949719c9f54958ce057db095fc1fb3295

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36785 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15720 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39037 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39426 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32949 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18273 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12835 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31513 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37347 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32507 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11590 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11639 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40675 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33410 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33130 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37516 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11879 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9698 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35637 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13540 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12275 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4802 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12779 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Nice using the gdk API for dmabuf handling.

carlosgcampos committed:
[GTK] Use the new GTK API to create a GdkTexture from a DMA-BUF buffer
https://bugs.webkit.org/show_bug.cgi?id=268445

Reviewed by Alejandro G. Castro.

Using GdkDmabufTextureBuilder we can create a GdkTexture to pass to GTK
without having to use EGL.

* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::create):
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::BufferDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::BufferDMABuf::didUpdateContents):
(WebKit::AcceleratedBackingStoreDMABuf::Renderer::snapshot const):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBuffer):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.h:

Canonical link: https://commits.webkit.org/273890@main
